server/src/main/java/org/elasticsearch/common/util/BytesRefArray.java

```diff
@@ -64,10 +65,7 @@ public BytesRefArray(StreamInput in, BigArrays bigArrays) throws IOException {
             // bytes
             long sizeOfBytes = in.readVLong();
             bytes = bigArrays.newByteArray(sizeOfBytes, false);
-
-            for (long i = 0; i < sizeOfBytes; ++i) {
-                bytes.set(i, in.readByte());
-            }
+            bytes.loadFrom(in);
```

Comment:
We can use the BytesRef iterator here instead of introducing a new method, but it seems unsafe if we materialize the returned BytesRef. We have an assertion, but not a hard failure for checking this condition.